static/js/components/AudioPlayer.js

```diff
@@ -0,0 +1,116 @@
+        {
+          album: this.props.title,
+          name:  this.props.description,
+          url:   this.props.url
+        }
+      ],
+      waveforms: {
+        sample_rate: 3000
+      }
+    })
+  }
+
+  componentWillUnmount() {}
+
+  backTenClick = () => {
+    Amplitude.skipTo(Amplitude.getSongPlayedSeconds() - 10, 0, null)
+  }
+
+  forwardThirtyClick = () => {
+    Amplitude.skipTo(Amplitude.getSongPlayedSeconds() + 30, 0, null)
+  }
+
+  seekBarClick = (e: Object) => {
+    const seekBar = this.seekBar && this.seekBar.current
+    if (seekBar && seekBar.getBoundingClientRect && seekBar.offsetWidth) {
+      const offset = seekBar.getBoundingClientRect()
+      const offsetWidth = seekBar.offsetWidth
+      const x = e.pageX - offset.left
+      Amplitude.setSongPlayedPercentage(
+        (parseFloat(x) / parseFloat(offsetWidth)) * 100
+      )
+    }
+  }
+
+  render() {
+    return (
+      <div className="audio-player-container-outer">
+        <div className="audio-player-container-inner">
+          <div className="audio-player-titles">
+            <div
+              className="audio-player-podcast-name"
+              data-amplitude-song-info="album"
+            />
+            <div
+              className="audio-player-podcast-title"
+              data-amplitude-song-info="name"
+            />
+          </div>
+          <div className="audio-player-controls">
+            <div className="audio-player-button" onClick={this.backTenClick}>
+              <span className="material-icons">replay_10</span>
+            </div>
+            <div
+              className="audio-player-button amplitude-play-pause"
+              id="play-pause"
+            >
+              <span className="material-icons" />
+            </div>
+            <div
+              className="audio-player-button"
+              onClick={this.forwardThirtyClick}
+            >
+              <span className="material-icons">forward_30</span>
+            </div>
+            <div className="audio-player-progress-container">
+              <time className="audio-player-time-text-left">
+                <span className="amplitude-current-minutes" />:
+                <span className="amplitude-current-seconds" />
+              </time>
+              <progress
+                className="amplitude-song-played-progress audio-player-progress"
+                id="song-played-progress"
+                ref={this.seekBar}
+                onClick={this.seekBarClick}
+              />
+              <time className="audio-player-time-text-right">
+                <span className="amplitude-duration-minutes" />:
+                <span className="amplitude-duration-seconds" />
+              </time>
+            </div>
+            <div className="audio-player-playback-speed-container">
+              <button className="light-outlined amplitude-playback-speed" />
+            </div>
+          </div>
+        </div>
+      </div>
+    )
+  }
+}
```

Comment:
one general comment: we're trying to mainly use functional components and react's [hooks](https://reactjs.org/docs/hooks-intro.html) API for new features. I think if you want to stick with a class-based component for this that's fine - but you may find the hooks APIs nicer to work with (I do!).